tests/NuGet.Services.AzureSearch.FunctionalTests/BasicTests/V2SearchProtocolTests.cs

```diff
@@ -0,0 +1,272 @@
+                .Data
+                .Any(x => IsPrerelease(x.Version));
+            Assert.True(hasPrereleaseVersions);
+
+            hasPrereleaseVersions = resultsWithoutPrerelease
+                .Data
+                .Any(x => IsPrerelease(x.Version));
+            Assert.False(hasPrereleaseVersions);
+        }
+
+        [Fact]
+        public async Task ResultsHonorSemverLevel()
+        {
+            var searchTerm = "packageId:" + Constants.TestPackageIdSemVer2;
+            var results = await V2SearchAsync(new V2SearchBuilder { Query = searchTerm, IncludeSemVer2 = true });
+
+            Assert.NotNull(results);
+            Assert.True(results.Data.Count >= 1);
+            var atleastOneResultWithSemVer2 = results
+                .Data
+                .Any(x => IsSemVer2(x.Version));
+
+            Assert.True(atleastOneResultWithSemVer2);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetSortByData))]
+        public async Task ResultsAreOrderedBySpecifiedParameter(string orderBy, Func<V2SearchResultEntry, object> GetPropertyValue, bool reverse = false)
+        {
+            var results = await V2SearchAsync(new V2SearchBuilder { Query = "json", SortBy = orderBy });
+
+            Assert.NotNull(results);
+            Assert.True(results.Data.Count > 1);
+
+            int count = results.Data.Count < 10 ? results.Data.Count : 10;
+            var topResults = results
+                .Data
+                .Select(GetPropertyValue)
+                .Take(count)
+                .ToList();
+
+            // Descending comparers
+            Func<object, object, bool> comparer = (x1, x2) => {
+                return topResults[0].GetType() == typeof(DateTime)
+                        ? DateTime.Compare((DateTime)x1, (DateTime)x2) >= 0
+                        : string.Compare(x1.ToString(), x2.ToString()) >= 0;
+                };
+
+            for (int i = 1; i < topResults.Count(); i++)
+            {
+                if (reverse)
+                {
+                    // flip for ascending comparison
+                    Assert.True(comparer(topResults[i], topResults[i-1]));
+                } else
+                {
+                    Assert.True(comparer(topResults[i - 1], topResults[i]));
+                }
+            }
+        }
+
+        [Fact]
+        public async Task ResultsMatchIdQueryFieldSearch()
+        {
+            var results = await V2SearchAsync(new V2SearchBuilder { Query = "packageId:" + Constants.TestPackageId });
+
+            Assert.NotNull(results);
+            Assert.True(results.Data.Count == 1);
+            Assert.Equal(Constants.TestPackageId, results.Data.First().PackageRegistration.Id, StringComparer.OrdinalIgnoreCase);
+        }
+
+        [Fact]
+        public async Task ResultsMatchAuthorQueryFieldSearch()
+        {
+            var results = await V2SearchAsync(new V2SearchBuilder { Query = "author:" + Constants.TestPackageAuthor });
```

Comment:
In the test output are the URLs traced? I find it useful in E2E tests to be able to click a URL of the failure and see the bad results right immediately